ReactWindows/ReactNative.Net46/Modules/WebSocket/ProxyHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿using ReactNative.Shared.Modules.WebSocket;
+using System;
+using System.Net;
+
+namespace ReactNative.Modules.WebSocket
+{
+    internal class ProxyHelper
+    {
+        public bool IsBypassed { get; private set; } = true;
+        public string ProxyAddress { get; private set; } = string.Empty;
+        public string UserName { get; set; } = string.Empty;
+        public string Password { get; private set; } = string.Empty;
+
+        public ProxyHelper(Uri address)
+        {
+            // we get wss:// (or ws://) scheme that is not all proxies handle correctly. For better autodiscovering
+            // replace wss:// to https:// (ws:// to http:// accordingly)
+
+            var normalizedAddress = new UriBuilder(address)
+            {
+                Scheme = address.Scheme == "wss" ? Uri.UriSchemeHttps : Uri.UriSchemeHttp
+            };
+
+            var proxy = WebRequest.DefaultWebProxy;
```

Comment:
pass the webProxy into the ctor as IWebProxy so this class can be easily unit tested with mocks.